examples/source_type.py

```diff
@@ -79,37 +80,58 @@ def sym_coord_bases(self):
                       for x, idx in zip(self.sym_coordinates,
                                         self.sym_coord_indices)])
 
-    def point2grid(self, u, m, t=t):
-        """Generates an expression for generic point-to-grid interpolation"""
+    @property
+    def index_matrix(self):
+        """Spatial indexing expressions for all grid points
+        contributing to spatial interpolation.
+        """
+        return [tuple([idx + ii + self.nbpml for ii, idx
+                       in zip(inc, self.sym_coord_indices)])
+                for inc in self.increments]
+
+    def point2grid(self, u, m, u_t=t, p_t=t):
```

Comment:
I really hope we have a plan to internalise these interpolations into the library code soon.